python/tank/deploy/tank_commands/validate_config.py

```diff
@@ -113,30 +246,139 @@ def _run(self, log):
         log.info("")
         log.info("")
         log.info("")
-        
-        
 
+    def _dump(self, log, envs, params):
+        """
+        Dump the supplied environments with the specified parameters
+
+        :param log: A logger instance.
+        :param envs: list of environment objects
+        :param params: parameter dict.
+        """
+
+        if params["dump_sparse"]:
+            dump_type = environment.UpdateAllSettingsFormat.SPARSE
+            dump_display_type = "sparse"
+        else:
+            dump_type = environment.UpdateAllSettingsFormat.FULL
+            dump_display_type = "full"
+
+        for env in envs:
+
+            # build the full output path from the dump location directory
+            # and the basename of the source environment file.
+            output_path = os.path.abspath(
+                os.path.join(params["dump_location"],
+                os.path.basename(env.disk_location))
+            )
+
+            # make sure the output path doesn't match the environment path.
+            # don't allow writing over existing environments.
+            if output_path == env.disk_location:
+                raise TankError(
+                    "Dump locaiton matches the existing environment path. "
+                    "Refusing to overwrite existing environment file."
+                )
+
+            # build a copy of the environment file that we can use to process.
+            (filename, ext) = os.path.splitext(
+                os.path.basename(env.disk_location)
+            )
+            tmp_env_path = os.path.join(
```

Comment:
Looking at the `WritableEnvironment` code again, it is very much tied to reading/writing files on disk (processing includes, etc). I don't see an easy way to add a clone method that returns an in-memory environment. I think what I'll do instead is change `make_full` and `make_sparse` to `dump_full` and `dump_sparse` respectively, each accepting a file like object. That seem like a reasonable compromise?